### PR TITLE
[4.0] Upgrade php-stemmer 2.0

### PR DESCRIPTION
**Pull request by HLeithner**
Pull Request for Issue #27329 .

### Summary of Changes
Upgrade php-stemmer from version 1.2 to 2.0, refactoring stemmer class to use stemmer factory.

### Testing Instructions
Use smart search with different languages and content.

### Expected result

Nothing changes.

### Actual result

Everything still works.

### CODE REVIEW

**Comment by brianteeman** [comment]
Did you mean to delete all those files?

**Comment by HLeithner** [comment]
Yes they are not longer needed.

**Comment by brianteeman** [comment]
thanks for the clarification

File: administrator/components/com_finder/Indexer/Language.php [hunk 1]
```diff
@@ -44,6 +48,43 @@ class Language
...
+	public function __construct($locale = null)
+	{
+		if ($locale !== null)
+		{
+			$this->language = $locale;
+		}
+
+		if ($this->language === null)
+		{
+			throw new Exception('Can\'t initial stemmer, locale is not set', 500);
```
Comment (mbabker) [inline administrator/components/com_finder/Indexer/Language.php]: Use a more specific Exception subclass.  IMO `throw new \Exception();` is a code smell as that makes anything trying to handle a specific possible exception turn into a catch for all unhandled exceptions (the aim should be that someone can catch specific exceptions they want to handle while letting other exceptions bubble up, using the root Exception class violates this).

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on a2e5c36674d9585b814d05c62b46f23072c37bdc<br /><br />Looks good. Cleared index and re-indexed successfully. Added multiple articles and categories and re-indexed. No errors and data indexed correctly.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27568">issues.joomla.org/tracker/joomla-cms/27568</a>.</sub>

**Comment by HLeithner** [comment]
Thx for the text but I will change the exception later but shouldn't effect your test.
@mbabker would assume * as default language betyer instead throwing an exception?

File: administrator/components/com_finder/Indexer/Language/El.php [hunk 2]
```diff
@@ -1,1013 +0,0 @@
...
- */
-
-namespace Joomla\Component\Finder\Administrator\Indexer\Language;
-
-defined('_JEXEC') or die;
-
-use Joomla\Component\Finder\Administrator\Indexer\Language;
-
-/**
- * Greek language support class for the Finder indexer package.
```
Comment (wilsonge) [inline administrator/components/com_finder/Indexer/Language/El.php]: I Don't see greek in the new version

File: administrator/components/com_finder/Indexer/Language/Zh.php [hunk 3]
```diff
@@ -1,79 +0,0 @@
...
-
-use Joomla\Component\Finder\Administrator\Indexer\Language;
-use Joomla\String\StringHelper;
-
-/**
- * Chinese (simplified) language support class for the Finder indexer package.
- *
- * @since  4.0.0
- */
-class Zh extends Language
```
Comment (wilsonge) [inline administrator/components/com_finder/Indexer/Language/Zh.php]: Or chinese

File: administrator/components/com_finder/Indexer/Language.php [hunk 4]
```diff
@@ -44,6 +48,43 @@ class Language
...
+		}
+
+		try
+		{
+			$this->stemmer = StemmerFactory::create($this->language);
+		}
+		catch (NotFoundException $e)
+		{
+			// We don't have a stemmer for the language
```
Comment (wilsonge) [inline administrator/components/com_finder/Indexer/Language.php:82]: We have two languages specific for Joomla - Greek + Chinese - We need to fallback to stuff in Joomla at this point then bail

Comment (HLeithner) [reply administrator/components/com_finder/Indexer/Language.php:82]: done and thanks

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**